kura/org.eclipse.kura.core.cloud/src/main/java/org/eclipse/kura/core/cloud/CloudPayloadProtoBufEncoderImpl.java

```diff
@@ -46,46 +49,40 @@ public CloudPayloadProtoBufEncoderImpl(KuraPayload kuraPayload) {
         KuraPayloadProto.KuraPayload.Builder protoMsg = KuraPayloadProto.KuraPayload.newBuilder();
 
         // set the timestamp
-        if (this.m_kuraPayload.getTimestamp() != null) {
-            protoMsg.setTimestamp(this.m_kuraPayload.getTimestamp().getTime());
+        if (this.kuraPayload.getTimestamp() != null) {
+            protoMsg.setTimestamp(this.kuraPayload.getTimestamp().getTime());
         }
 
         // set the position
-        if (this.m_kuraPayload.getPosition() != null) {
+        if (this.kuraPayload.getPosition() != null) {
             protoMsg.setPosition(buildPositionProtoBuf());
         }
 
         // set the metrics
-        for (String name : this.m_kuraPayload.metricNames()) {
+        for (final Map.Entry<String, Object> entry : this.kuraPayload.metrics().entrySet()) {
+            final String name = entry.getKey();
+            final Object value = entry.getValue();
 
             // build a metric
-            Object value = this.m_kuraPayload.getMetric(name);
             try {
-                KuraPayloadProto.KuraPayload.KuraMetric.Builder metricB = KuraPayloadProto.KuraPayload.KuraMetric
-                        .newBuilder();
+                KuraMetric.Builder metricB = KuraMetric.newBuilder();
                 metricB.setName(name);
 
                 boolean result = setProtoKuraMetricValue(metricB, value);
                 if (result) {
-                    metricB.build();
-
```

Comment:
Because it is not needed. In this context, not using the return value, it is a no-op.